### PR TITLE
OkHttp3 Connection Builder

### PR DESCRIPTION
**Pull request by commonsguy**
This is the first of the HTTP stack integrations that are for stacks outside the Android SDK. I made some decisions that you need to review. I will apply the same pattern that gets accepted here for the other two stacks.

- **Module directory**: `libnetcipher-okhttp3` is the new module, mostly to be able to pull in the dependency (`com.squareup.okhttp3:okhttp:3.2.0`) without requiring all NetCipher users to use it

- **Module package name**: `info.guardianproject.onionkit.okhttp3` (so, `.okhttp3` appended to the main NetCipher module's package name)

- **Test location**: `androidTest/` sub-project under `libnetcipher-okhttp3`, just to cut back the number of top-level directories

- **Test package name**: `info.guardianproject.onionkit.okhttp3.test`

- Java code package: `info.guardianproject.netcipher.client`, so basically this module just adds another class to the same Java package that is used for the `HttpURLConnection` and common code

----

Other notes:

- OkHttp3 has a `minSdkVersion` of 9, so I set that on the `libnetcipher-okhttp3` module itself as well.

- Most likely, you are going to need to tweak the `build.gradle` file for the `libnetcipher-okhttp3` module to tie into your distribution process. When these pull requests are done, you will have a total of four artifacts to distribute: `libnetcipher` itself plus artifacts for OkHttp3, Volley, and HttpClient integrations. Those latter three will depend upon the `libnetcipher` artifact. Right now, I am just using `compile project(':libnetcipher')` to accomplish that, which works when you have the whole repo checked out. You will need to see how your POMs come out for these other modules and whether they have an appropriate dependency on the right `libnetcipher` version. The approach that I take in my CWAC modules is to use `debugCompile project('...')` (to pull in the local module for development) and `releaseCompile '...'` (pointing to the artifact for release builds). However, that is tied to how I publish my AARs and may not be suitable for your distribution approach.

### CODE REVIEW

**Comment by eighthave** [comment]
The package name should be based on `info.guardianproject.netcipher` everywhere.  If you find any existing references to `info.guardianproject.onionkit`, we should rename those.

Looks like the test with Orbot installed failed:
```
info.guardianproject.netcipher.StrongConnectionBuilderTest > testOrbotInstalled[test(AVD) - 5.0.2] FAILED 

	junit.framework.AssertionFailedError: we did not get an Orbot status

	at info.guardianproject.netcipher.StrongConnectionBuilderTest.testOrbotInstalled(StrongConnectionBuilderTest.java:97)

info.guardianproject.netcipher.StrongConnectionBuilderTest > testStrongConnectionBuilder[test(AVD) - 5.0.2] FAILED 

	junit.framework.AssertionFailedError: we did not get an Orbot status

	at info.guardianproject.netcipher.StrongConnectionBuilderTest.testStrongConnectionBuilder(StrongConnectionBuilderTest.java:115)
```


About the yahoo.com hostname failure, I have an issue in with Travis-CI https://github.com/travis-ci/travis-ci/issues/6044

File: libnetcipher-okhttp3/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="info.guardianproject.onionkit.okhttp3"
```
Comment (eighthave) [inline libnetcipher-okhttp3/AndroidManifest.xml]: should be `package="info.guardianproject.netcipher.okhttp3"`

File: libnetcipher-okhttp3/androidTest/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="info.guardianproject.onionkit.okhttp3.test"
```
Comment (eighthave) [inline libnetcipher-okhttp3/androidTest/AndroidManifest.xml]: should be `package="info.guardianproject.netcipher.okhttp3.test"`

**Comment by eighthave** [comment]
the code looks good, just fix the onionkit packageName and hopefully the Orbot installed test

**Comment by commonsguy** [comment]
@eighthave The package names are updated per your request. With regards to the tests, I [commented on this](https://github.com/guardianproject/NetCipher/pull/47#issuecomment-219174963) and have not yet received a reply.

**Pull request merged**